.travis.yml

```diff
@@ -1,76 +1,100 @@
-
-
-dist: xenial
 sudo: required
 language:
   - generic
+compiler:
+  - gcc
 cache:
   - apt
+services:
+  - docker
 
 env:
   global:
     - ROS_DISTRO=kinetic
     - ROS_CI_DESKTOP="`lsb_release -cs`"  # e.g. [precise|trusty|...]
-    - CI_SOURCE_PATH=$(pwd)
-    - ROSINSTALL_FILE=$CI_SOURCE_PATH/dependencies.rosinstall
-    - CATKIN_OPTIONS=$CI_SOURCE_PATH/catkin.options
     - ROS_PARALLEL_JOBS='-j8 -l6'
-    # Set the python path manually to include /usr/-/python2.7/dist-packages
-    # as this is where apt-get installs python packages.
     - PYTHONPATH=$PYTHONPATH:/usr/lib/python2.7/dist-packages:/usr/local/lib/python2.7/dist-packages
+    - BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi)
+
 
 ################################################################################
 
-# Install system dependencies, namely a very barebones ROS setup.
 before_install:
-  - sudo rm /var/lib/dpkg/lock
-  - sudo sh -c "echo \"deb http://packages.ros.org/ros/ubuntu $ROS_CI_DESKTOP main\" > /etc/apt/sources.list.d/ros-latest.list"
-  - sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
-  - sudo apt-get update -qq
-  - sudo apt-get install -y python-catkin-pkg python-rosdep python-wstool ros-$ROS_DISTRO-catkin python-catkin-tools
-  - source /opt/ros/$ROS_DISTRO/setup.bash
-  # Prepare rosdep to install dependencies.
-  - sudo rosdep init
-  - rosdep update
+  - docker run -e BRANCH -e PYTHONPATH -d --name alica-test -v $TRAVIS_BUILD_DIR:/travis ros:kinetic-ros-core-xenial tail -f /dev/null
+  - docker ps
 
-# Create a catkin workspace with the package under integration.
 install:
-  - mkdir -p ~/catkin_ws/src
-  - cd ~/catkin_ws/src
-  - catkin_init_workspace
-  # Create the devel/setup.bash (run catkin_make with an empty workspace) and
-  # source it to set the path variables.
-  - cd ~/catkin_ws
-  - catkin build
-  - source devel/setup.bash
-  # Add the package under integration to the workspace using a symlink.
-  - cd ~/catkin_ws/src
-  - ln -s $CI_SOURCE_PATH .
+  - docker exec -t alica-test bash -c "
+    rm /etc/ros/rosdep/sources.list.d/20-default.list && 
+    rosdep init && 
+    rosdep update && 
+    mkdir -p ~/catkin_ws/src && 
+    cd ~/catkin_ws/src && 
+    apt-get update && 
+    apt-get install -y 
+        python-catkin-pkg 
+        python-rosdep 
+        python-wstool 
+        python-catkin-tools 
+        ros-kinetic-catkin 
```

Comment:
Might be better to change all kinetic references to $ROS_DISTRO

Reply:
done